gensim/models/base_any2vec.py

```diff
@@ -330,12 +334,18 @@ def __init__(self, sentences=None, workers=3, vector_size=100, epochs=5, callbac
                 self.neg_labels[0] = 1.
 
         if sentences is not None:
-            if isinstance(sentences, GeneratorType):
+            if multistream:
+                if not isinstance(sentences, (tuple, list)):
+                    raise TypeError("If multistream=True, you must pass tuple or list as the sentences argument.")
+                if any(isinstance(stream, GeneratorType) for stream in sentences):
+                    raise TypeError("You can't pass a generators as input streams. Try an iterator.")
```

Comment:
Minor language fix: `a generator`

Reply:
No, here I explicitly check that `streams` (elements of `sentences` list) are not generators.

Reply:
@jayantj means that it should be 'generator' singular rather than 'generators' plural.